### PR TITLE
ConfigDialog: fix bug in DoubleSpinWidget default values

### PR DESCRIPTION
**Pull request by hius07**
Thanks https://github.com/koreader/koreader/pull/9558#issuecomment-1264447739.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9580)

### CODE REVIEW

**Review by poire-z — APPROVED** [review]
This fixes the crash for me.

**Comment by poire-z** [comment]
Is it expected that we don't show any "Default value: xyz" when the user has not set any ? Even though there are defaults values: our shipped default values ?

**Comment by hius07** [comment]
This behaviour is the same as on long-pressing the name of the setting. We show "Default value: not set" even we have shipped the defaults. So, there are "Default value set by a user" and "Default value shipped by us".

**Comment by poire-z** [comment]
OK.
(Although I guess it could be nice to have that button with our shipped defaults - when a user has messed with settings and ends up being unhappy, it's good to be able to get back to the shipped defaults - even if you're not happy with them: you'll be as not happy as everybody, which might be a small relief :)

**Comment by hius07** [comment]
> it could be nice to have that button with our shipped defaults

Always or only if the user has not set his own default values?

**Comment by poire-z** [comment]
Whereever this PR is showing that "Default values:" button:
- if the user has not set his own values: show our shipped ones
- if the user has set his own values: show his values (and assume having set some as defaults, he knows what he's doing and don't need to be reminded about our shipped defaults)

(Otherwise, it would need yet another button :) or may be long-press on that "Default values" would reset them to our shipped defaults. But that might be overkill and add kludge in the code. So, not really necessary.)

**Comment by hius07** [comment]
> * if the user has not set his own values: show our shipped ones
> * if the user has set his own values: show his values

Done

File: frontend/ui/widget/configdialog.lua [hunk 1]
```diff
@@ -577,15 +577,15 @@ function ConfigOption:init()
                                 self.options[c].more_options_param.show_true_value_func = self.options[c].show_true_value_func
                             end
                             Notification:setNotifySource(Notification.SOURCE_BOTTOM_MENU_MORE)
-                            local default_value
+                            local default_value_local
```
Comment (poire-z) [inline frontend/ui/widget/configdialog.lua:580]: `local default_value_local` sounds odd :)
May be it should be `current_default_value(s)` ?

**Comment by NiLuJe** [comment]
Looking at ConfigDialog kinda makes makes cross-eyed, so thanks for dealing with this, I definitely like the idea ;).

**Pull request merged**